player/lua/ytdl_hook.lua

```diff
@@ -57,6 +59,35 @@ local function edl_escape(url)
     return "%" .. string.len(url) .. "%" .. url
 end
 
+local function time_to_secs(time_string)
+    if not time_string then
+        return nil
+    end
+
+    local secs = 0
+    string.gsub(time_string, "[%d.]+", function (number)
+        secs = secs * 60 + tonumber(number)
+    end)
+
+    return secs
+end
+
+local function extract_chapters(data, video_length)
+    local ret = {}
+
+    for line in data:gmatch("[^\r\n]+") do
+        local time_string = string.match(line, "([%d:.]+:[%d.]+)")
```

Comment:
`%d{1,2}(:%d{2})+` would be (close to?) ideal as it would not backtrack, but that won't match `00:4:04`.

In javascript, `/(?:\d{1,2}:){1,2}\d{2}/` would match exactly what youtube matches in the example screenshot. Maybe it works by just replacing the `\` with `%`?

Note that it does backtrack, but only if the timestamp actually is followed by a colon.